src/main/java/com/couchbase/lite/store/SQLiteViewStore.java

```diff
@@ -308,9 +380,10 @@ public void emit(Object key, Object value) {
                         String[] args = {Long.toString(sequence), keyJson, valueJson};
                         store.getStorageEngine().execSQL(queryString(
                                 "INSERT INTO 'maps_#' (sequence, key, value) VALUES(?,?,?)"), args);
+                        insertedCount.incrementAndGet();
                     } catch (Exception e) {
                         Log.e(Log.TAG_VIEW, "Error emitting", e);
-                        // find a better way to propagate this back
+                        throw new RuntimeException(e);
```

Comment:
I just worried RuntimeException could not be caught by anything.
``
RuntimeException and its subclasses are unchecked exceptions. Unchecked exceptions do not need to be declared in a method or constructor's throws clause if they can be thrown by the execution of the method or constructor and propagate outside the method or constructor boundary
```